mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -61,20 +67,70 @@ class LogisticRegressionModel (
 
   override protected def predictPoint(dataMatrix: Vector, weightMatrix: Vector,
```

Comment:
The argument to the gradient calculation is properly a vector of weights, so that need not change for API reasons. So is it just having to do the translation? it's a line of code I think, although requires a copy. Maybe someone else can weigh in with an opinion too.